### PR TITLE
Re-add system-probe kitchen test tweaks

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Re-add https://github.com/DataDog/datadog-agent/pull/15165 but disabling timeouts specifically for `pkg/network/tracer`
This is the expected outcome:
* All packages will have a default timeout of 10min
* Packages under `pkg/network/protocols` will have a stricter limit of 5 min
* `pkg/network/tracer` will have timeouts disabled for now

### Motivation

It was my belief that we were relying on the default package test timeout of 10min, but my previous PR had the unintended side-effect of going from _no timeouts_ to then 10min timeouts instead. This is due to how we generate kitchen tests.
For example:

```
func TestTimeout(t *testing.T) {
	fmt.Println("now", time.Now())
	deadline, ok := t.Deadline()
	fmt.Println("deadline", deadline, ok)
}
```
running this test using the go test tool without setting a timeout will produce the following output:
```
go test . -test.v -test.v
=== RUN   TestTimeout
now 2023-01-20 11:57:07.273809225 -0500 EST m=+0.000164095
deadline 2023-01-20 12:07:07.273786194 -0500 EST m=+600.000141067 true
--- PASS: TestTimeout (0.00s)
PASS
ok  	example.com	0.001s
```

but if we generate an executable test file (like we do for CI/kitchen tests):
```
$ go test . -c -o testsuite
```
and then execute it
```
$ ./testsuite -test.v -test.count=1
=== RUN   TestTimeout
now 2023-01-20 11:58:49.817109119 -0500 EST m=+0.000546454
deadline 0001-01-01 00:00:00 +0000 UTC false
--- PASS: TestTimeout (0.00s)
PASS
```
we can see that timeouts are disabled by default

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

File: test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb [hunk 1]
```diff
@@ -20,6 +20,32 @@
...
+
+DEFAULT_TIMEOUT = "10m"
+
+def get_timeout(package)
+  match_size = 0
+  timeout = DEFAULT_TIMEOUT
+
+  # determine longest match
+  TIMEOUTS.each do |k, v|
+    k = Regexp.new(k)
```
Comment (brycekahle) [inline test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb:37]: nit: you could probably move `Regexp.new` into the dictionary definition, so it is only done once.

File: test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb [hunk 2]
```diff
@@ -20,6 +20,32 @@
...
+    v = String(v)
```
Comment (brycekahle) [inline test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb:38]: why is this necessary? isn't it already a string?

Comment (p-lambert) [reply test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb:38]: just being defensive since values are not strongly typed and someone could add a number or something else in the hash declaration

Comment (brycekahle) [reply test/kitchen/test/integration/system-probe-test/rspec/system-probe-test_spec.rb:38]: 👍
